lib/dynflow/director/sequence_cursor.rb

```diff
@@ -65,23 +64,18 @@ def done_here?
 
       def next_steps
         move if @no_error_so_far
-        if done?
-          if @parent_cursor
-            return @parent_cursor.what_is_next(self, @no_error_so_far)
-          else
-            return []
-          end
+        return steps_todo unless done?
+        if @parent_cursor
+          return @parent_cursor.what_is_next(self, @no_error_so_far)
         else
-          return steps_todo
+          return []
         end
       end
 
       def add_todo(flow)
         case flow
         when Flows::Sequence
-          @todo << SequenceCursor.new(@flow_manager, flow, self).tap do |cursor|
-            cursor.move
-          end
+          @todo << SequenceCursor.new(@flow_manager, flow, self).tap(&:move)
```

Comment:
Funny thing, this two things are not the same, as the `&:` variant has issues wiht protected method. I will revert this commit in my PR, as I don't want to risk to hit this unexpected behaviour changes in alredy pretty big PR
```
SequenceCursor.new(@flow_manager, flow, self).tap(&:move)
# NoMethodError: protected method `move' called for #<Dynflow::Director::SequenceCursor:0x0055555fbc9b70>

```